init.c

```diff
@@ -934,5 +941,13 @@ int luaopen_libcutorch(lua_State *L)
   lua_pushlightuserdata(L, state);
   lua_setfield(L, -2, "_state");
 
+  THCState **thc_pointer = (THCState**)lua_newuserdata(L, sizeof(void*));
+  *thc_pointer = state;
+  lua_newtable(L);
+  lua_pushcfunction(L, &cutorch_shutdown);
+  lua_setfield(L, -2, "__gc");
+  lua_setmetatable(L, -2);
+  lua_setfield(L, -2, "_stategc");
```

Comment:
Would you mind adding some comments explaining how this works? I always get confused when it comes to reading Lua C API code :)